### PR TITLE
Frontend Homework 10

### PR DESCRIPTION
**Pull request by asish918**
## Drift Video
[Drift](https://video.drift.com/v/abqscYvWIW2DnZA4EF7TcoIHRg5rDSdA2HUpqQMiNOGuc)

## Component Diagram
Todo-Component

### Desktop View
Screenshot 2024-02-20 at 20 44 15

Screenshot 2024-02-20 at 20 44 36

Screenshot 2024-02-20 at 20 44 59

### Mobile View
Screenshot 2024-02-20 at 20 44 28

Screenshot 2024-02-20 at 20 44 44

Screenshot 2024-02-20 at 20 45 07

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 2 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: frontend/homework/10/src/components/navbar/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Task } from '../../types';
+import './Navbar.scss'
+
+interface NavbarProps {
+    searchState: string;
+    setSearchState: React.Dispatch<React.SetStateAction<string>>;
+    setSearchResults: React.Dispatch<React.SetStateAction<Task[]>>;
+    tasks: Task[];
+}
```
Comment (Nitesh2503) [inline frontend/homework/10/src/components/navbar/Navbar.tsx:9]: Add the interfaces in the different file

File: frontend/homework/10/src/components/navbar/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export default function Navbar({ searchState, setSearchState, setSearchResults, tasks }: NavbarProps) {
+    function searchChangeHandler(event: React.ChangeEvent<HTMLInputElement>): void {
+        if (event.target.value.length === 0) {
```
Comment (Nitesh2503) [inline frontend/homework/10/src/components/navbar/Navbar.tsx:13]: Could be directly written as `!event.target.value.length`

File: frontend/homework/10/src/components/navbar/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        setSearchResults(tasks.filter(item => {
+            const lowerCaseItem = item.name.toLowerCase();
+            return lowerCaseItem.includes(searchState.toLowerCase());
+        }));
+    }
+
+    return (
+        <nav className="navbar">
+            <h1 className="navbar__heading">Item Lister</h1>
+
```
Comment (Nitesh2503) [inline frontend/homework/10/src/components/navbar/Navbar.tsx:33]: Format the code properly

File: frontend/homework/10/src/components/todo/Todo.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    searchState: string;
+    searchResults: Task[];
+}
+
+
+export default function Todo({ tasks, setTasks, searchState, searchResults }: TodoProps) {
+    const [taskInput, setTaskInput] = useState('');
+
+    function addTask() {
+        if (taskInput.trim().length === 0 || taskInput.length === 0) return;
```
Comment (Nitesh2503) [inline frontend/homework/10/src/components/todo/Todo.tsx:17]: The code could be optimized

File: frontend/homework/10/src/components/todo/Todo.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import { Task } from '../../types'
+import './Todo.scss'
+
+interface TodoProps {
+    tasks: Task[];
+    setTasks: React.Dispatch<React.SetStateAction<Task[]>>;
+    searchState: string;
+    searchResults: Task[];
+}
+
```
Comment (Nitesh2503) [inline frontend/homework/10/src/components/todo/Todo.tsx:11]: Follow the same for this too

File: frontend/homework/10/src/components/todo/Todo.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                            {result.name}
+                        </div>
+                    ))
+                }
+                {
+                    searchResults.length === 0 && searchState.length > 0 &&
+                    <div>
+                        No Results
+                    </div>
+                }
```
Comment (Nitesh2503) [inline frontend/homework/10/src/components/todo/Todo.tsx:64]: Use functions to make it more readable

**Comment by Nitesh2503** [comment]
- Initially when no items are there then show the proper message there
- Nice work on the UI
- The submit button must be disabled initially and the UI should also look alike

**Pull request closed without merging**